### DISCUSSION TITLE
Compilation for Apple Silicon

### DISCUSSION THREAD

**Discussion by thinhhoang95** [GAA-UAM/scikit-fda#354] [Ideas]
Given that M1 Macs are now becoming more widely available on the market, I think it could be helpful to compile scikit-fda on ARM64 to work with Apple Silicon SoCs. It seems that scikit-learn is already available on M1, I wonder if scikit-fda actually depends on any dependency that has yet to be ported to M1 so that library compilation can eventually take place?

**Comment by vnmabus** [comment]
Thank you for your suggestion!

For now, scikit-fda is pure Python. But we depend in several libraries that are partially written in C, such as fdasrsf. Maybe you should try to take a look at those. Also, I am not sure M1 Mac is currently available in Github actions for testing, which makes more difficult to ensure that it works.

**Reply by thinhhoang95** [reply]
Thanks for the reply

I currently have an M1 Mac so I might try to build scikit-fda's dependency on ARM64 to see if it works. I'll let you know about the progress.
